src/Text/Layout/Table/Primitives/Table.hs

```diff
@@ -10,32 +13,38 @@ import           Text.Layout.Table.Spec.Util
 
 
 -- | Draw a horizontal line that will use the delimiters around the
--- appropriately and visually separate by 'hSpace'.
+-- appropriately and visually separate by 'hSpace', or Nothing if
+-- all delimiters are null.
 hLineDetail
     :: StringBuilder b
-    => Char -- ^ The space character that is used as padding.
-    -> Char -- ^ The delimiter that is used on the left side.
-    -> Char -- ^ The delimiter that is used in between cells.
-    -> Char -- ^ The delimiter that is sued on the right side.
+    => String -- ^ The space characters that are used as padding.
+    -> String -- ^ The delimiter that is used on the left side.
+    -> String -- ^ The delimiter that is used in between cells.
+    -> String -- ^ The delimiter that is used on the right side.
     -> Row b -- ^ A row of builders.
-    -> b -- ^ The formatted line as a 'StringBuilder'.
-hLineDetail hSpace delimL delimM delimR cells =
-    mconcat $ intersperse (charB hSpace) $ charB delimL : intersperse (charB delimM) cells ++ [charB delimR]
+    -> Maybe b -- ^ The formatted line as a 'StringBuilder', or Nothing if all delimiters are null.
+hLineDetail ""     ""     ""     ""     = const Nothing
+hLineDetail hSpace delimL delimM delimR = Just . hLineDetail' hSpace delimL delimM delimR
 
--- | A simplified version of 'hLineDetail' that will use the same delimiter
--- for everything.
-hLine
+-- | Draw a horizontal line that will use the delimiters around the
+-- appropriately and visually separate by 'hSpace'.
```

Comment:
Please make sure the sentences make sense.  (I'm afraid some of those errors I did myself.  I think what I meant was probably `around the content appropriately`.)